garak/generators/nim.py

```diff
@@ -174,4 +183,180 @@ def _prepare_prompt(self, prompt):
         return text
 
 
+class NVMultimodal(Generator):
+    """Wrapper for text + image / audio to text NIMs. Expects NIM_API_KEY environment variable.
+
+    Expects keys to be a dict with keys 'text' (required), and 'image' or 'audio' (optional).
+    Message is sent with 'role' and 'content' where content is structured as text
+    followed by <img> and/or <audio> tags ala https://build.nvidia.com/microsoft/phi-4-multimodal-instruct
+    """
+
+    DEFAULT_PARAMS = Generator.DEFAULT_PARAMS | {
+        "uri": "https://integrate.api.nvidia.com/v1/chat/completions",
+        "retry_5xx": True,
+        "suppressed_params": {"n", "frequency_penalty", "presence_penalty", "stop"},
+        "max_input_len": 180_000,
+        "ratelimit_codes": [429],
+        "skip_codes": [],
+        "request_timeout": 60,
+        "proxies": None,
+        "verify_ssl": True,
+        "max_tokens": 512,
+        "top_p": 0.70,
+        "stream": False,
+    }
+
+    ENV_VAR = "NIM_API_KEY"
+    generator_family_name = "NVMultimodal"
+    modality = {"in": {"text", "image", "audio"}, "out": {"text"}}
+    uri = "https://integrate.api.nvidia.com/v1/chat/completions"
+
+    def __init__(self, name="", config_root=_config):
+        self._load_config(config_root)
+        self.name = name
```

Comment:
If `name` is passed and set before calling `_load_config()` the value is held vs overridden by `config_root` provided values, however if name is passed as `""` which matches the `default` signature for the parameter it will be set replaced if `config_root["generators"]["nim"]["name"]` exists.
```suggestion
        self.name = name
        self._load_config(config_root)
```